src/test/java/rx/SingleTest.java

```diff
@@ -879,6 +879,19 @@ public void doOnSuccessShouldNotSwallowExceptionThrownByAction() {
     }
 
     @Test
+    public void doOnSubscribeShouldInvokeAction() {
+        Action0 action = mock(Action0.class);
+        Single<Integer> single = Single.just(1).doOnSubscribe(action);
```

Comment:
@artem-zinnatullin That's a good idea! Just added `doOnSubscribeShouldInvokeActionBeforeSubscriberSubscribes` test.